grakn-graql/src/main/java/ai/grakn/graql/internal/analytics/MedianVertexProgram.java

```diff
@@ -129,7 +129,7 @@ public void storeState(final Configuration configuration) {
     public void loadState(final Graph graph, final Configuration configuration) {
         super.loadState(graph, configuration);
         configuration.subset(RESOURCE_TYPE).getKeys().forEachRemaining(key ->
-                statisticsResourceLabelIds.add(LabelId.of(configuration.getInt(RESOURCE_TYPE + "." + key))));
+                statisticsResourceLabelIds.add((LabelId) configuration.getProperty(RESOURCE_TYPE + "." + key)));
```

Comment:
Also if we create our own `configuration ` wrapper we can return the object as they should be. No need for dirty castings everywhere. But that's a future concern I think.